### PR TITLE
[IntSet] Add an implementation of encoding/decoding IntSet's to/from the IFT sparse bit set encoding.

### PR DESCRIPTION
**Pull request by garretrieger**
https://w3c.github.io/IFT/Overview.html#sparse-bit-set-decoding

### CODE REVIEW

**Comment by rsheeter** [comment]
Is this IFT specific - in which case possibly it belongs in IFT's codebase - or general purpose?

File: int-set/Cargo.toml [hunk 1]
```diff
@@ -7,6 +7,7 @@ description = "A fast sparse and invertible bit set for u32's. Port of harfbuzz'
 
 [dependencies]
 font-types = { version = "0.5.5", path = "../font-types"}
+thiserror = "1.0.58"
```
Comment (rsheeter) [inline int-set/Cargo.toml:10]: I think we don't use this in fontations and probably shouldn't here either, just implement the error manually

**Review by rsheeter — APPROVED** [review]
Looks reasonable enough, LGTM once CI is happy.

File: int-set/src/input_bit_stream.rs [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+    /// Reads the next two bits and returns the corresponding branch factor.
+    ///
+    /// See: <https://w3c.github.io/IFT/Overview.html#sparse-bit-set-decoding>
+    /// Returns None if the stream does not have enough remaining bits.
+    pub fn read_branch_factor(&mut self) -> Option<u8> {
+        let bit_0 = self.read_bit()?;
+        let bit_1 = self.read_bit()?;
+
+        match (bit_1, bit_0) {
```
[the remark this answers is not in the document]
Comment (garretrieger) [reply int-set/src/input_bit_stream.rs:26]: This is set up to match the decoding table in the spec which uses the terminology bit 0 and bit 1. If I put the read's inline then the bit order will be reversed from what's in the spec's table.

**Pull request merged**